python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -607,6 +634,8 @@ def build_op_context(
     config: Any = None,
     partition_key: Optional[str] = None,
     mapping_key: Optional[str] = None,
+    run: Optional[DagsterRun] = None,
```

Comment:
cc @dpeng817 

I forget what our past discussions around adding run were - if we had a real reason to not or were just waiting for a good reason to

Reply:
I might just switch this to use a crazy magic mock instead

Reply:
DagsterRun is an api that we expect users not to construct directly, only use for type annotations. Given that often in test scenarios they would be mocking this object, it seemed like a pretty weird API surface area to add